lib/plugins/backup.rb

```diff
@@ -33,28 +33,57 @@ class Backup
 
       def initialize
         yield self if block_given?
-        raise ArgumentError if @bucket.nil? || @files.nil? || @hooks.nil?
+        raise ArgumentError if @bucket.nil? || @files.nil? || @files.empty? || @hooks.nil?
+        @target_name ||= '%{app_name}_%{timestamp}_%{user}.tar.gz'
+      end
+
+      # Factory method to create preconfigured
+      # Backup plugins. Uploads current template
+      # and parameter files.
+      # @param backup [String] target bucket name
+      # @return [Backup] configured backup object
+      def self.to_bucket(bucket:)
+        raise ArgumentError if bucket.nil?
+        Moonshot::Plugins::Backup.new do |b|
+          b.bucket = bucket
+          b.files = [
+            { path: %w(cloud_formation), name: '%{app_name}.json' },
+            { path: %w(cloud_formation parameters), name: '%{stack_name}.yml' }
+          ]
+          b.hooks = [:post_create, :post_update]
+        end
       end
 
-      def backup(resources)
+      # Main worker method, creates a tarball of the given
+      # files, and uploads to an S3 bucket.
+      # @param resources [Resources] injected Moonshot resources
+      def backup(resources) # rubocop:disable Metrics/AbcSize
+        raise ArgumentError if resources.nil?
+
         @app_name = resources.stack.app_name
         @stack_name = resources.stack.name
-        @target_name ||= '%{app_name}_%{timestamp}_%{user}.tar.gz'
-
-        tar_out = tar @files
-        zip_out = zip tar_out
-        upload zip_out
-      rescue StandardError => e
-        raise e
-      ensure
-        tar_out.close unless tar_out.nil?
-        zip_out.close unless zip_out.nil?
+        @target_name = render(@target_name)
+
+        resources.ilog.start log_message.clone << ' in progress.' do |s|
+          begin
+            tar_out = tar @files
```

Comment:
We generally use parenthesis for method invocation in this codebase, best to be consistent in this method.